### ISSUE TITLE
Negative numbers cause unexpected start of atom errors

### ISSUE THREAD

**Issue by healeycodes** [thesephist/ink#5]
I couldn't find anything specific about negative numbers in `SPEC.md` or the documentation so: I am making the assumption that the minus sign `-` should be used instead of the UnaryOp `~` when declaring negative numbers.

Please do correct me if this is wrong!

## Bug

Platforms: macOS and Linux
Version: v1.7

<br>

Using negative numbers cause an `unexpected start of atom` error.

Declaring as type number:
```
std := load('std')
log := std.log

a := -1
log(a)
` unexpected start of atom, found '-' [4:7] `
```

Declaring within a composite value (via array syntax):
```
std := load('std')
log := std.log

numbers := [1, -1]
log(numbers)
` unexpected start of atom, found '-' [4:17] `
```

Declaring within a composite value (via map syntax):
```
std := load('std')
log := std.log

numbers := {
	a: -1
}
log(numbers)
` unexpected start of atom, found '-' [5:7] `
```

Passing as an argument to a function:

```
std := load('std')
log := std.log

log(type(-1))
` unexpected start of atom, found '-' [4:11] `
```

<br>

Also occurs in the REPL:

```
> -1
syntax error: unexpected start of atom, found '-' [1:2]
```

<br>

## Temporary workaround

Minus the number from zero as it's declared.

e.g. `0-23`, `0-8`

Or use the UnaryOp.

e.g. `~23`, `~8`
```
std := load('std')
log := std.log

a := 0-1
b := ~2
log(a)
log(b)
` -1
  -2  `

```

**Comment by thesephist** [comment]
Thanks for the report @healeycodes! This is intended behavior, and the tilde character is intended to be used to make negative number literals, though it's admittedly a little idiosyncratic and I guess insufficiently documented. Based on your comment I added a note in the website under [Values and operators](https://dotink.co/docs/overview/#values-and-operators) to note this, because `~` and `-` are hard to distinguish visually.

When I first designed the syntax I thought it'd be neat if there was one token/operator used both for negating booleans, as well as for negating numbers, and I picked `~`. But in hindsight I think it defies convention enough that it's hard to notice. I find that I got used to it pretty quickly, but something I'm not completely fixed on if it becomes a hassle -- I wouldn't be completely opposed to making `-` take its place in the syntax because it shouldn't break any current code. I'll throw it in my pile of "ink things to reconsider if I make significant language changes"

```
~true `` -> false
~2 `` -> negative 2
```
